cpp/tests/utilities/column_utilities.cu

```diff
@@ -459,23 +458,15 @@ class corresponding_rows_not_equivalent {
     }
   };
 
-  cudf::row_equality_comparator<cudf::nullate::YES> comp;
-
   __device__ bool operator()(size_type index)
   {
     auto const lhs_index = lhs_row_indices.element<size_type>(index);
     auto const rhs_index = rhs_row_indices.element<size_type>(index);
 
+    cudf::experimental::row::equality::nan_equal_physical_equality_comparator comp;
     if (not comp(lhs_index, rhs_index)) {
-      auto lhs_col = this->d_lhs.column(0);
-      auto rhs_col = this->d_rhs.column(0);
-      return type_dispatcher(lhs_col.type(),
-                             typed_element_not_equivalent{},
-                             lhs_col,
-                             rhs_col,
-                             lhs_index,
-                             rhs_index,
-                             fp_ulps);
+      return type_dispatcher(
+        lhs.type(), typed_element_not_equivalent{}, lhs, rhs, lhs_index, rhs_index, fp_ulps);
```

Comment:
This is no longer needed. `typed_element_not_equivalent` was needed to compare `NaN`. Now `comp` can do that.

Reply:
I meant we also can remove `typed_element_not_equivalent` completely.

Reply:
Humnn, probably we still need it to compare floating point numbers that are "very close".